pymc/smc/smc.py

```diff
@@ -570,3 +547,86 @@ def _logp_forw(point, out_vars, in_vars, shared):
     f = compile_pymc([inarray0], out_list[0])
     f.trust_input = True
     return f
+
+
+class Particles:
+    """
+    List of draws from variables (particles)
+    belonging to a model that can be modified
+    in place as if they were an array.
+    """
+
+    @classmethod
+    def build(cls, draws: int, initial_point, init_rnd, model, dual_representation=False):
```

Comment:
Why is this not part of the `__init__` method?

Reply:
Because by using a factory we are decoupling details of a very particular way of building an object, from the object itself. For example, Particles shouldn't need an initial point to be built, it's just one way of doing it, by splitting this two things we can build Particles with the dependencies it needs, and no circunstancial objects.

Reply:
I don't get it yet. Do you have a case in mind where you will make use of this split?